crates/transaction-pool/src/metrics.rs

```diff
@@ -0,0 +1,36 @@
+//! Transaction pool metrics.
+
+use metrics::{describe_counter, register_counter, Counter};
+
+pub(crate) struct TxPoolMetrics {
```

Comment:
We technically have multiple subpools, so perhaps this should be scoped by subpool? I.e. one for "ready" transactions, one where the basefee is too low, and one last I don't remember. Even more so, it is possible to have as many custom subpools as you want I think (cc @mattsse) - so really, these metrics should be scoped by subpool name

Reply:
hmm, I think we should definitely have counters for the entire thing, and perhaps additional counters for subpools.
so starting with these is fine imo.